packages/account/src/Sections/Profile/LanguageSettings/language-settings.tsx

```diff
@@ -1,78 +1,51 @@
-                            <DesktopWrapper>
-                                <FormSubHeader title={localize('Select Language')} />
-                            </DesktopWrapper>
-                            <div className='settings-language__language-container'>
-                                {allowed_language_keys.map(language_key => {
-                                    return (
-                                        <LanguageRadioButton
-                                            key={language_key}
-                                            id={language_key}
-                                            language_code={language_key}
-                                            is_current_language={values.language_code === language_key}
-                                            name='language-radio-group'
-                                            onChange={(event: React.ChangeEvent<HTMLInputElement>) =>
-                                                setFieldValue('language_code', event.target.value)
-                                            }
-                                        />
-                                    );
-                                })}
-                            </div>
-                        </div>
-                        <FormFooter>
-                            <Button
-                                className='account-form__footer-btn'
-                                type='submit'
-                                data-testid={'submit-button'}
-                                has_effect
-                                text={localize('Submit')}
-                                large
-                                primary
-                                is_disabled={isCurrentLanguage(values.language_code)}
-                            />
-                        </FormFooter>
-                    </form>
-                );
-            }}
-        </Formik>
+        <div className='settings-language'>
+            <FormSubHeader title={localize('Select Language')} />
+            <div className='settings-language__language-container'>
+                {allowed_language_keys.map(language_key => {
+                    return (
+                        <LanguageRadioButton
+                            key={language_key}
+                            id={language_key}
+                            language_code={language_key}
+                            is_current_language={current_language === language_key}
+                            name='language-radio-group'
+                            onChange={async () => {
+                                changeSelectedLanguage(language_key);
+                                await i18n.changeLanguage?.(language_key);
+                            }}
+                        />
+                    );
+                })}
+            </div>
+        </div>
     );
 };
 
 export default connect(({ common }: TCoreStore) => ({
```

Comment:
@yauheni-deriv Out of your card scope but we can remove the connect if you are willing to 😈

Reply:
yes, I'd like to do it.. but accounts package has no implemented provider yet..we can’t use useStore without provider, right?

Reply:
@yauheni-deriv Oh you are right, My bad 🙇🏻 Please ignore this comment 🙏🏻